modules/ai/assets/js/editor/hooks/use-prompt.js

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+import { setStatusFeedback } from '../api';
+
+const normalizeResponse = ( { text, response_id: responseId, usage } ) => {
+	const creditsData = usage ? ( usage.quota - usage.usedQuota ) : 0;
+	const credits = Math.max( creditsData, 0 );
+
+	return {
+		result: text,
+		responseId,
+		credits,
+	};
+};
+
+const usePrompt = ( fetchData, initialState ) => {
+	const [ isLoading, setIsLoading ] = useState( false );
+	const [ error, setError ] = useState( '' );
+	const [ data, setData ] = useState( initialState );
+
+	const send = async ( ...args ) => {
+		setError( '' );
+		setIsLoading( true );
+
+		fetchData( ...args )
+			.then( ( result ) => setData( normalizeResponse( result ) ) )
+			.catch( ( err ) => setError( err?.responseText || err ) )
+			.finally( () => setIsLoading( false ) );
+	};
+
+	const sendUsageData = () => data.responseId && setStatusFeedback( data.responseId );
+
+	const reset = () => {
+		// The credits should not be reset, because the previous value is still valid for the next request.
+		setData( ( prevState ) => ( { ...prevState, result: '', responseId: '' } ) );
```

Comment:
maybe? just to make it more clear

```suggestion
		setData( ( prevState ) => ( { credits: prevState.credits, result: '', responseId: '' } ) );
```

or even:
```suggestion
		setData( ( { credits } ) => ( { 
			credits,
			result: '', 
			responseId: '' 
		} ) );
```

Reply:
Done.